docs/changelog.rst

```diff
@@ -10,18 +10,19 @@ Changes will be added here periodically from the "Suggested changelog entry"
 block in pull request descriptions.
 
 
-
 IN DEVELOPMENT
 --------------
 
 Changes will be summarized here periodically.
 
-Version 2.10.1 (Oct 2?, 2022)
+Version 2.10.1 (Oct 24, 2022)
 -----------------------------
 
+There is some concern about ABI compatibility between 2.10.x (including this release) and previous versions, possibly related to cross-module exception handling. This is still developing and tracked under `#4105 <https://github.com/pybind/pybind11/pull/4105>`_. Please review if this is important to you before upgrading, and report an issue or comment on an issue if there's a problem.
```

Comment:
I think it's best to delete this paragraph completely now. PR #4283 demonstrates that there could be a problem, but it is such an unusual corner case, I think it will be more confusing than helpful to mention it here.